### PR TITLE
Justinian Plague and epidemic diseases #96

### PR DESCRIPTION
**Pull request by rquinio**
- Activate epidemic diseases by extending start_date to 1.1.1 as they were already present in the antiquity (except measles - 500 A.D.).
- Add the Justinian plague as a 1 time occurrence of  bubonic plague in the 6th century. Note: was discussed [in the old forum thread](https://forum.paradoxplaza.com/forum/index.php?threads/when-the-world-stopped-making-sense-a-480-migrational-period-mod.772971/page-172#post-17635909).
- Recurrence of the plague in the 7th and 8th century.
- Add narrative events to notify player when:
  - Plague appears in the world
  - Plague appears in the realm
  - Religious head is contaminated, lowering moral authority (note: may not fire if character dies too quickly)

I was able to test by manually changing outbreak_chance from 0.01 to 1, and loading a 6th century bookmark.

### CODE REVIEW

**Comment by loup99** [comment]
Looks good. Are there any events related to vanilla plagues that canbe reused in the mod?

**Comment by CaudilloFranco** [comment]
I researched this a bit and found that there were extreme weather events that are tied in to the Justinian Plague and the crisis and chaos surrounding it. There was widespread crop failure and such. Would it be relevant to add less wealth and more cold climates to all provinces on the map, if possible? There was an eruption of a volcano on a Pacific island that blocked out the sun and made a Volcanic Winter.

**Comment by loup99** [comment]
We plan on changing the climate according to Numhar's map, but through events I don't think it is possible.

**Comment by CaudilloFranco** [comment]
Ok. I only meant adding province modifiers to the whole map, not making events but I do like the idea of changing the climate a lot.

**Comment by rquinio** [comment]
Oh interesting, I never heard about these [extreme weather events of 535-536](https://en.wikipedia.org/wiki/Extreme_weather_events_of_535%E2%80%93536) !
It seem to be coincidence that it happened 5 years before the Justinian Plague, no ?
Wikipedia article says  David Keys theory of connections between weather event, the plague, rise of Islam, etc.. is "not widely accepted".

Maybe it could be a rare narrative event that puts a modifier (city_tax_modifier / supply_limit / ...) on every province for 2 years ?

**Comment by CaudilloFranco** [comment]
Sorry about that, I thought I was replying to you, didn't know that it would update YOUR comment haha. But Ya, I really like that idea for the province modifiers, but perhaps it can last a year or two more due to the huge changes it made to the Earth in general? It's your call though.

**Comment by loup99** [comment]
I think agree on that such hasty conclusions linking events together are not verycredible. Lookslike a coincidence. 

A narrative event seems appropriate, but there was some discussion around natural catastrophes as even if they can't be snowballed, I think it is veryboring to know that each time in your game you will have the same catastrophe at the same point of time.

**Comment by rquinio** [comment]
@loup99 Good point. Good news is, as it reuses the has_bubonic_plague illness trait, vanilla events already take into account in event modifiers.
But there are no specific vanilla events about bubonic plague, except for event id = 6114 (health_events.txt) that simply makes the character survive the disease without much details, so it's fine.

There's one mod that adds few events to the Black Death (http://steamcommunity.com/sharedfiles/filedetails/?id=436833455) but it's ahistorical (zombies, ...).

What I could add is a narrative event to notify players a terrible disease has appeared and is spreading ?

**Comment by CaudilloFranco** [comment]
I completely understand, maybe it didn't cause the Plague or affect it in any way, but it stil had a lasting impact on the environment. I know this probably isn't relevant to the subject but in Total War Attila, the Germanic or otherwise Barbarian tribes are migrating south due to a combination of the Huns invading and a cold climate settling over Northernmost Europe, causing a domino effect essentially, pushing everyone towards the Romans. 

In-game, there are already migrations like this in the sense that they have CBs to migrate to another territory that they have conquered, so a modifier like this could (not a guarantee) replicate the tensions and problems created with tribes knocking on your door and squeezing the more southern kingdoms until they finally give in.

**Comment by loup99** [comment]
@rquinio I think it would be appropriate to have a narrative event, it makes it a lot more interesting to follow. 

As for the environment, another narrative event would be appropriate.

**Comment by rquinio** [comment]
Events added (I've updated PR description), please review !

In one of my tests, plague actually started from the ERE under Justinian. If as the player he would get the plague and survive, that could be a nice achievement ;)
Then it spreaded to whole Europe.

In another test it started in Arabia, the Coptic Pope was contaminated, but the plague was somehow contained in that area.

**Comment by CaudilloFranco** [comment]
@rquinio The events look great to me! They all look very thorough. If you guys need help, chances are I can aid with this. Only if you need help though.

**Comment by loup99** [comment]
Looks really good!

**Comment by rquinio** [comment]
@CaudilloFranco Sure, things can be enriched in the future, this is just a basis.
Comments on PR are locked once it is merged, so I've created issue #96 to track this topic: you can suggest ideas, give feedback or implement stuff yourself !

File: WTWSMS/common/disease/WTWSMS_disease.txt [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+# Initial outbreak
+justinian_plague = {
+	contagiousness = 0.95
+	outbreak_chance = 0.01
+	effect = {
+		city_tax_modifier = -0.8
+		supply_limit = -4
+		max_attrition = 0.05
+	}
```
Comment (CaudilloFranco) [inline WTWSMS/common/disease/WTWSMS_disease.txt]: Could we add an extra revolt risk penalty, or is that too hard for the player otherwise?

Comment (rquinio) [inline WTWSMS/common/disease/WTWSMS_disease.txt]: Is there historical evidence of peasant revolts during such plagues ?
I'm afraid the revolt stack would get crushed by attrition.

**Comment by CaudilloFranco** [comment]
There were many instances of revolts during the Middle Ages due to poor conditions like Plague and Famine which happened during the Crisis of the Late Middle Ages. I know there is bound to be a few notable revolts because I know I've read it before somewhere. Here is a good source for more info: https://en.m.wikipedia.org/wiki/Popular_revolt_in_late_medieval_Europe

**Comment by rquinio** [comment]
Ok, as I understand the plagues caused serious labour shortage, increasing the bargaining power of the peasantry, which with resistance of the nobility would trigger revolts.
But I believe this happened in the aftermath, not while the epidemics was still active, so I don't think modifier should be directly attached to the epidemics (which lasts only 2-3 months in-game).

> Justinian showed no mercy towards the ruined freeholders. Even then, he did not refrain from demanding the annual tax, not only the amount at which he assessed each individual, but also the amount for which his deceased neighbors were liable

Based on this, I was thinking of some event for realms struck by the plague, where the liege would have the choice to keep same taxing on the peasantry (increasing global revolt risk, ...), or show mercy but face a general tax decrease for several years.

**Comment by CaudilloFranco** [comment]
I'm really liking that idea a lot. Who would the event concern? The top liege? All the vassals? Will burghers (mayors) and bishops have this choice too?

**Comment by loup99** [comment]
Good idea! But no specific events for the mayors and bishops, they are never player controlled.

**Comment by CaudilloFranco** [comment]
It has been a while since anyone has commented on this: is there any progress besides this so far? Just curious.

**Comment by rquinio** [comment]
Well, I haven't progressed on the taxing/revolt event(s).
The PR is tested and 'complete' in regard to its description, so we could merge this, and add the event(s) later.

**Pull request merged**